### PR TITLE
Fix mitobim form

### PR DESCRIPTION
**Pull request by neoformit**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [x] This PR updates an existing tool or tool collection

Minor update to Mitobim - fields marked as optional in the tool form are actually required by mitobim, this has caused a lot of failed jobs for our users.

### CODE REVIEW

File: tools/mitobim/mitobim.xml [hunk 1]
```diff
@@ -56,13 +56,13 @@
...
-	</param>
+        <param argument="--ref" type="text" value="" label="Reference ID" help="If resuming, use the same as in previous iteration/initial MIRA assembly" optional="false">
+            <validator type="regex" message="&quot;, ', and / are not allowed in IDs">[^'\"/]*$</validator>
+        </param>
         <param argument="--readpool" type="data" format="fastq,fastq.gz" label="Read pool" help="Read pairs need to be interleaved for full functionality of the '-pair' option below"/>
-	<param argument="--sample" type="text" value="" label="Sample ID" help="If resuming, the sampleID needs to be identical to that of the previous iteration / MIRA assembly">
-            <validator type="regex" message="&quot;, &apos;, and / are not allowed in IDs">[^'\"/]*$</validator>
-	</param>
+        <param argument="--sample" type="text" value="" label="Sample ID" help="If resuming, the sampleID needs to be identical to that of the previous iteration / MIRA assembly" optional="false">
+            <validator type="regex" message="&quot;, ', and / are not allowed in IDs">[^'\"/]*$</validator>
```
Comment (bgruening) [inline tools/mitobim/mitobim.xml:67]: ther is any non-empty validator that you can add.

Comment (bgruening) [reply tools/mitobim/mitobim.xml:67]: you could also add a test with an empty input and specify that the test is expected to fail

**Review by bgruening — COMMENTED** [review]
@neoformit thanks! Please bump the tool version number.

File: tools/mitobim/mitobim.xml (hunk 1 above)
Comment (bernt-matthias) [reply tools/mitobim/mitobim.xml:67]: optional actually makes no sense for text parameters, the validator would be required

Comment (neoformit) [reply tools/mitobim/mitobim.xml:67]: >optional actually makes no sense for text parameters

It's unfortunate that `optional="true"` still results in the tool form label showing "optional" on those fields:

image

Does it really make no sense to hide this for required text fields? Maybe from a "galaxy backend" sense, yes, but from a UX perspective..?

In any case, I guess the validator will at least stop the tool from being executed, but I'm sure it'll still confuse the user that it says "optional" and then "sorry, not actually optional".

Comment (neoformit) [reply tools/mitobim/mitobim.xml:67]: And I should add that `validator type="empty_field"` appears to do nothing when I run in planemo serve (despite many tools using the same pattern), so let's hope it does something on real Galaxy because otherwise this PR has changed nothing?

**Comment by mvdbeek** [comment]
Deployment status: **success**

https://github.com/galaxyproject/tools-iuc/actions/runs/21517593643

**Comment by bernt-matthias** [comment]
> That empty-field is not working for you is concerning, it should. Lets see ...

In my test it did not work because there was still an `optional="true"` .. which I removed here https://github.com/galaxyproject/tools-iuc/pull/7628/commits/6e06ebaef3a7a4940758a1a2f696009d5abe5780 .. then everything was fine.

**Comment by neoformit** [comment]
Ahhh thanks @bernt-matthias
